Removed the * from secret searches that end with the search character…

angeorgieff opened the pull request:
… of _, -, or ., as Slack doesn't return results when the wildcard is applied after these special characters. This

now supports Slack, Dropbox, and Github tokens.

angeorgieff committed:
Removed the * from secret searches that end with the search character of _, -, or ., as Slack doesn't return results when the wildcard is applied after these special characters. This
now supports Slack, Dropbox, and Github tokens.

angeorgieff commented:
Great project! I found that the slack tokens, github tokens, and dropbox tokens we're showing up in the secret's results due to the * following their special characters of `_` `-` or `.`, so it'll now remove the * if these characters are detected.

NoF0rte commented on .gitignore:
```diff
@@ -1,2 +1,5 @@
 .slack-slurp.yaml
+*.txt
+*.json
+.gitattributes
```
Remove these three additions from the PR

NoF0rte commented:
I think I somewhat understand the issue here. Slack's `*` searching is just partial word searching and I assume in Slack's eyes the characters `-`, `_`, and `.` stop the partial word matching or something like that. Is it just for whenever those characters show up within the search term or if those characters are at the end?

NoF0rte commented:
Also, please remove the `main` binary from the PR.

angeorgieff committed:
removed binary and updated regex for asterisk decision to only check the last character of the keyword for ., -, and _

angeorgieff commented:
Great point about the regex. I updated the regex so it only checks the last character in the keyword instead of anywhere in the keyword. Also removed the binary.

NoF0rte commented on pkg/slurp/slurp.go:
```diff
@@ -315,7 +315,12 @@ func (s Slurper) GetSecretsAsync(detectrs ...detectors.Detector) (chan SecretRes
+				if !re.MatchString(keyword) {
+					searchString = searchString + "*"
+				}
+				messageChan, err2Chan := s.SearchMessagesAsync(fmt.Sprintf("%s", searchString))
```
I am a little picky when it comes to the style so I apologize. The following is how I would do this, the first line would be on line 314:
```go
nonStarSearchableRe := regexp.MustCompile(`(-|\.|_)$`)
for _, detector := range selectedDetectors {
	var err error
	keywords := detector.Keywords()
	for _, keyword := range keywords {
		if !nonStarSearchableRe.MatchString(keyword) {
			keyword = keyword + "*"
		}
		messageChan, err2Chan := s.SearchMessagesAsync(keyword)
```

I know that technically `searchString` better describes what is being passed to `s.SearchMessagesAsync()` but I don't think we need to create a brand new variable just for that.

NoF0rte reviewed (changes requested):
Address the feedback I made for the different files. Other than that, looks good to me. Thanks and good find!

angeorgieff committed:
updated structure and removed unnecessary variable declaration, gitignore reverted

angeorgieff commented:
Agreed on the structural change. Thanks for the awesome repo!
